Gems/Multiplayer/Code/Source/Pipeline/NetworkPrefabProcessor.cpp

```diff
@@ -125,12 +120,11 @@ namespace Multiplayer
 
         auto&& [object, networkSpawnable] =
             ProcessedObjectStore::Create<AzFramework::Spawnable>(uniqueName, context.GetSourceUuid(), AZStd::move(serializer));
```

Comment:
Instead of creating the spawnable here, consider pushing the PrefabDOM for the .network.spawnable down the pipeline instead. This has two benefits: first the `PrefabCatchmentProcessor` will be able to resolve entities, entity aliases and spawnable links. The second benefit is that any future processors will be able to be applied to .network.spawnables as well. The first benefit may also remove the need for `FindOrCreateCachedInstanceAliasForPrefab`.

Reply:
Didn't understand the suggestion fully. So, when should the network spawnable be created? Because in this same function, the network entities are updated with the network spawnable asset id. If the network spawnable is not created, how can that updating take place?

Reply:
Yes, we need to have net spawnable asset ID here. 
Even if we pre-generated the asset ID and pushed the PrefabDom back into context, each of them would still have its own Instance (i.e. that'd be 2 separate calls into `PrefabCatchmentProcessor::ProcessPrefab`) and without `FindOrCreateCachedInstanceAliasForPrefab` the entity IDs would be random, so cross-instance entity references would still be broken.

Reply:
The Prefab Processing Stack starts with a single prefab but during processing additional prefabs can be added to the context using `AddPrefab`. For this case it would be a second Prefab with the network information. Note that adding a Prefab also requires giving it a unique name (which is also used as part of product name) such as "network".
When setting up an entity alias it's possible to pass in the target spawnable with `EntityAliasSpawnableLink` or the name of the prefab `EntityAliasPrefabLink`. For this case the latter can be used when adding an entity alias to the root spawnable with the prefab name "network".
At the bottom of the stack is the `PrefabCatchmentProcessor`. This will go over all Prefabs and turn them into spawnables. Part of this process is resolving the entity aliases from prefab name to the spawnable asset id. (See `ResolveSpawnableEntityAliases`.)

Add additional prefabs to the stack instead of going straight for the spawnables has other benefits as well because more stack nodes can eventually work on those prefabs. Think for instance how a tiling system can be added transparently (and how it would behave differently at runtime depending if it was inserted before or after the network node in the stack). Of course the biggest benefit is that the work needed to handle the conversion from a prefab to a spawnable, such as resolving the asset ids, are already solved by the `PrefabCatchmentProcessor` plus future updates would automatically be applied to the .network.spawnable. (Though please consider that the spawnable entity aliases isn't battle tested yet, so there may still be bugs in resolving those asset ids.)